### PR TITLE
Enable ajax navigation for cfr changes.

### PR DESCRIPTION
**Pull request by jmcarp**
* Use <doc_id>-<doc-type>-<section> format for section IDs
* Add helper to parse IDs

rdzq1ku72o

### CODE REVIEW

File: regulations/static/regulations/js/source/helpers.js [hunk 1]
```diff
@@ -287,5 +287,24 @@ module.exports = {
     toggleExpandable: function($expandable, dur) {
         $expandable.toggleClass('open')
           .next('.chunk').slideToggle(dur);
+    },
+
+    /**
+     * Parse a preamble section ID to a path array.
+     * @param {string} id Preamble section ID
```
Comment (cmc333333) [inline regulations/static/regulations/js/source/helpers.js]: It'd be helpful to include some examples here

File: regulations/views/preamble.py [hunk 2]
```diff
@@ -56,11 +56,20 @@ def get(self, request, *args, **kwargs):
...
 
-        context = {'sub_context': context, 'sub_template': template,
-                   'preamble': preamble, 'doc_number': label_parts[0]}
+        context = {
+            'sub_context': context,
+            'sub_template': template,
+            'preamble': preamble,
+            'doc_number': label_parts[0],
+            'section_prefix': prefix,
+            'markup_id': context['node']['label_id']
```
Comment (cmc333333) [inline regulations/views/preamble.py]: Just a heads up that this should probably be `context['node']['markup_id']`

**Comment by cmc333333** [comment]
I think it'll be a bit easier for future devs to understand what's going on if there's _less_ logic in the templates. Is there an easy way to refactor this to construct identifiers in Python and just pass them through to the template?

File: regulations/views/preamble.py (hunk 2 above)
Comment (jmcarp) [inline regulations/views/preamble.py]: TY

File: regulations/static/regulations/js/source/helpers.js (hunk 1 above)
Comment (jmcarp) [inline regulations/static/regulations/js/source/helpers.js]: :+1:

**Comment by jmcarp** [comment]
@cmc333333: would you recommend adding a new key in `PreambleHTMLBuilder` / `CFRHTMLBuilder`?

**Comment by cmc333333** [comment]
@jmcarp I think that'd work. I wonder what'd break if `markup_id` always included the the prefix? We could add it to the `PreambleHTMLBuilder` constructor, for example.

**Comment by jmcarp** [comment]
@cmc333333: I fixed the typo in the example, but the change isn't showing up here (although it's correct at https://github.com/jmcarp/regulations-site/commits/ajax-cfr-changes). Once this patch is in good shape, merging from git should pull in the example fix.

**Pull request merged**